libraries/SITL/SIM_SilentWings.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  simulator connector for ardupilot version of SilentWings
+*/
+
+#include "SIM_SilentWings.h"
+
+#include <stdio.h>
+
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+using namespace SITL;
+
+static const struct {
+    const char *name;
+    float value;
+    bool save;
+} sim_defaults[] = {
+    { "AHRS_EKF_TYPE", 10 },
+    { "INS_GYR_CAL", 0 },
+    { "EK2_ENABLE", 0 },
+    { "ARSPD_ENABLE", 1 },
+    { "ARSPD_USE", 1 },
+    { "INS_ACC2OFFS_X",    0.001 },
+    { "INS_ACC2OFFS_Y",    0.001 },
+    { "INS_ACC2OFFS_Z",    0.001 },
+    { "INS_ACC2SCAL_X",    1.001 },
+    { "INS_ACC2SCAL_Y",    1.001 },
+    { "INS_ACC2SCAL_Z",    1.001 },
+    { "INS_ACCOFFS_X",     0.001 },
+    { "INS_ACCOFFS_Y",     0.001 },
+    { "INS_ACCOFFS_Z",     0.001 },
+    { "INS_ACCSCAL_X",     1.001 },
+    { "INS_ACCSCAL_Y",     1.001 },
+    { "INS_ACCSCAL_Z",     1.001 },
+};
+
+SilentWings::SilentWings(const char *home_str, const char *frame_str) :
+    Aircraft(home_str, frame_str),
+    last_data_time_ms(0),
+    first_pkt_timestamp_ms(0),
+    time_base_us(0),
+    sock(true),
+    home_initialized(false),
+    inited_first_pkt_timestamp(false)
+{
+    // Try to bind to a specific port so that if we restart ArduPilot
+    // SilentWings keeps sending us packets. Not strictly necessary but
+    // useful for debugging
+    sock.bind("127.0.0.1", 6060);
```

Comment:
this means you can't run silentwings and ardupilot on different computers

Reply:
Right, will fix in the upcoming iteration.